stl/inc/format

```diff
@@ -4034,337 +3190,160 @@ _NODISCARD constexpr const _CharT* _Parse_range_specs(
+            _Throw_format_error("Range-types 's' and '?s' require type T to be charT.");
         }
-    }
-};
-
-template <class _Rng, class _CharT>
-concept _Const_formattable_range =
-    _RANGES input_range<const _Rng> && formattable<_RANGES range_reference_t<const _Rng>, _CharT>;
-
-template <class _Rng, class _CharT>
-using _Fmt_maybe_const = conditional_t<_Const_formattable_range<_Rng, _CharT>, const _Rng, _Rng>;
-
-template <range_format _Kind, _RANGES input_range _Rng, class _CharT>
-struct _Range_default_formatter;
-
-template <_RANGES input_range _Rng, class _CharT>
-struct _Range_default_formatter<range_format::sequence, _Rng, _CharT> {
-private:
-    using _Range_type = _Fmt_maybe_const<_Rng, _CharT>;
-
-    range_formatter<remove_cvref_t<_RANGES range_reference_t<_Range_type>>, _CharT> _Underlying;
-
-public:
-    constexpr void set_separator(const basic_string_view<_CharT> _Sep) noexcept {
-        _Underlying.set_separator(_Sep);
-    }
 
-    constexpr void set_brackets(
-        const basic_string_view<_CharT> _Opening, const basic_string_view<_CharT> _Closing) noexcept {
-        _Underlying.set_brackets(_Opening, _Closing);
-    }
+        break;
 
-    template <class _ParseContext>
-    constexpr _ParseContext::iterator parse(_ParseContext& _Ctx) {
-        return _Underlying.parse(_Ctx);
+    default:
+        _STL_UNREACHABLE; // can't happen, we've handled all possible outputs from _Parse_range_specs()
+        break;
     }
 
-    template <class _FormatContext>
-    _FormatContext::iterator format(_Range_type& _Elems, _FormatContext& _Ctx) const {
-        return _Underlying.format(_Elems, _Ctx);
+    if (_Specs._No_brackets) {
+        _Opening_bracket = basic_string_view<_CharT>{};
+        _Closing_bracket = basic_string_view<_CharT>{};
     }
-};
-
-template <_RANGES input_range _Rng, class _CharT>
-struct _Range_default_formatter<range_format::map, _Rng, _CharT> {
-private:
-    using _Map_type     = _Fmt_maybe_const<_Rng, _CharT>;
-    using _Element_type = remove_cvref_t<_RANGES range_reference_t<_Map_type>>;
 
-    range_formatter<_Element_type, _CharT> _Underlying;
+    return _Ctx.begin() + (_It - _Ctx._Unchecked_begin());
+}
 
-public:
-    constexpr _Range_default_formatter()
-        noexcept(is_nothrow_default_constructible_v<range_formatter<_Element_type, _CharT>>) /* strengthened */ {
-        static_assert(_Is_two_tuple<_Element_type>, "the element type of the formatted range must be either pair<T, U> "
-                                                    "or tuple<T, U> (N4981 [format.range.fmtmap]/1)");
+template <class _Ty, class _CharT, _RANGES input_range _Range, class _FormatContext>
+void _Range_formatter_format_as_sequence(const formatter<_Ty, _CharT>& _Underlying, _Range&& _Rng,
+    const basic_string_view<_CharT> _Separator, const basic_string_view<_CharT> _Opening_bracket,
+    const basic_string_view<_CharT> _Closing_bracket, _FormatContext& _Ctx) {

```

Comment:
The parameters of `_Range_formatter_format_as_sequence` are interleaving the data members with the arguments, which is confusing and inconsistent with the other member functions that are being extracted. We should consistently put the data members first.

Also occurs in `_Range_formatter_format` and `_Fill_align_and_width_formatter_format`.

(As an aside, I've pinged *again* about the modules bug VSO-1236041 that I reported back in Oct 2020 which is causing the trouble here.)